### ISSUE TITLE
yuedajiong-question #02: same or different encoding space?

### ISSUE THREAD

**Issue by yuedajiong** [facebookresearch/ijepa#7]
what is the essence of predictor?  can we use same encoders both input-side and target-side?

my understanding:
1) Lecun and this MidoAssran's implementation:
   pixel space  -> f_encode_input -> encoded-but-before-predict space #S_e -> f_predict -> predicted space #S_p
   pixel space ->f_ebncode_target -> encoded-for-loss-with-predicted space #S_t
   based on my limited understanding, the distance between S_t and S_p is nearer than the distance  between S_t and S_e.
2) Another:
   pixel space  -> f_encode_input -> encoded-but-before-predict space #S_e -> f_predict -> predicted space #S_p
   'FORCE' to use same f_encode_input as encoder_target (of course, need code modification).  (enen, we can use another reconsturction even multi-task based model to pre-train the encoder, we just need to train predicttor here.)
   so, this will be: the predictor prdicts masked information in SAME laten space.  few parameters, especial hierarchical encoders.

is this still reasonable for min/max information content: 
https://scontent-hkt1-1.xx.fbcdn.net/v/t39.2365-6/274349327_328573009069618_3830687132316417744_n.jpg?_nc_cat=110&ccb=1-7&_nc_sid=ad8a9d&_nc_ohc=laxyPF2DPaUAX-t_SPX&_nc_ht=scontent-hkt1-1.xx&oh=00_AfDrZb2-IiXgN7GjAyo0DCvL8qhV5-W0Jc6wdnzs-AyPJw&oe=648F1A69

**Comment by MidoAssran** [comment]
Hi @yuedajiong ,

1. For the first question, if you were to directly match the output of the context and target encoder, you would learn invariant representations that collapse the masked and unmasked representations to the same point. However, the idea behind I-JEPA is that we want to be able to predict masked regions from visible regions, not collapse them together. Hence, the predictor is essential for computing this mapping.
2. For the second question, this is definitely valid. You can pretrain the visual encoder and then just train the predictor in this frozen latent space, but the idea behind I-JEPA is to learn to *represent* the world by predicting it, hence we were interested in using this approach to actually train a visual encoder. You could of course still use the same network with unfrozen weights for the context and target, but you would need an explicit regularization to prevent collapse (e.g., using the information maximization terms in the diagram you pointed out). This would be an interesting research questions!
